logstash-core/src/main/java/org/logstash/ackedqueue/Queue.java

```diff
@@ -162,33 +163,45 @@ public void open() throws IOException {
         for (int pageNum = headCheckpoint.getFirstUnackedPageNum(); pageNum < headCheckpoint.getPageNum(); pageNum++) {
```

Comment:
hmm. after trying to force this to become an issue... I can't. Since purging of tail files only happens on ack... a checkpoint would also be created. There would only be an issue if something killed the process.. in which case, anything can be corrupt